.github/workflows/release.yml

```diff
@@ -41,77 +35,99 @@ on:
 
 jobs:
   release:
-    name: 'Release'
+    name: "Release"
     timeout-minutes: 30
     runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: write
     outputs:
       version: ${{ steps.get-version.outputs.version }}
       sha: ${{ steps.get-sha.outputs.sha }}
     env:
       ORG_GRADLE_PROJECT_nexusUsername: ${{ secrets.nexus-username }}
       ORG_GRADLE_PROJECT_nexusPassword: ${{ secrets.nexus-password }}
-      AWS_ACCESS_KEY_ID: ${{ secrets.aws-access-key }}
-      AWS_SECRET_ACCESS_KEY: ${{ secrets.aws-secret-key }}
-      AWS_REGION: ${{ inputs.aws_region }}
+      AWS_REGION: eu-west-1
     steps:
-      - name: 'Getting repository name'
+      - name: "Getting repository name"
         run: echo "GITHUB_REPO_NAME=${GITHUB_REPOSITORY#*/}" >> $GITHUB_ENV
-      - name: 'Print action context'
+
+      - name: "Print action context"
         run: |
           echo "Repository: ${{ env.GITHUB_REPO_NAME }}"
           echo "$GITHUB_CONTEXT"
         env:
           GITHUB_CONTEXT: ${{ toJson(github) }}
-      - name: 'Checkout'
-        uses: actions/checkout@v2
+
+      - name: "Checkout"
+        uses: actions/checkout@v4
         with:
           fetch-depth: 0
-      - name: 'Login to Amazon ECR' #Amazong Elastic Container Registry login action -> https://github.com/aws-actions/amazon-ecr-login
-        if: inputs.docker_image == true
+
+      - name: Configure AWS credentials
```

Comment:
This step is not needed when used in repos without Docker Images. Better to keep `if: inputs.docker_image == true`